com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -1358,5 +1400,13 @@ void DecideAction()
             m_Info.CopyActions(actions);
             m_ActuatorManager.UpdateActions(actions);
         }
+
+        internal void SetMultiAgentGroup(IMultiAgentGroup multiAgentGroup)
+        {
+            // Unregister from current group if this agent has been assigned one before
```

Comment:
I think if you want the agent to unregister from all groups, you should allow passing a null multiAgentGroup here. This would be better as something like
``` csharp
if(multiAgentGroup == null)
{
  m_GroupId == 0;
}
else
{
  var newId = multiAgentGroup.GetId();
  if(m_GroupId == 0 || m_GroupId == newId)
  {
    // Idempotent, so setting the same group has no effect
    m_GroupId = newId;
  }
  else 
  {
    throw new UnityAgentsException("Agent is already registered with a group. Unregister it first.");
  }
}
```

Reply:
I made it internal and only being called in MultiAgentGroup.RegisterAgent so I don't see there's any cases that it can be passed in a null group. Also it shouldn't be unregistered by just overwriting m_GroupId otherwise there would be delegates left in Agent.UnregisterFromGroup.

I can do throwing an exception if a user trying to assign an agent from one group directly to another group without unregistering from the first, instead of call unregister for them.